cmd/promtool/backfill_test.go

```diff
@@ -0,0 +1,441 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"io/ioutil"
+	"math"
+	"os"
+	"sort"
+	"testing"
+
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/storage"
+	"github.com/prometheus/prometheus/tsdb"
+	"github.com/stretchr/testify/require"
+)
+
+type backfillSample struct {
+	Timestamp int64
+	Value     float64
+	Labels    labels.Labels
+}
+
+func createTemporaryOpenMetricsFile(t *testing.T, text string) string {
+	newf, err := ioutil.TempFile("", "")
+	require.NoError(t, err)
+
+	_, err = newf.WriteString(text)
+	require.NoError(t, err)
+	require.NoError(t, newf.Close())
+
+	return newf.Name()
+}
+
+func sortSamples(samples []backfillSample) {
+	sort.Slice(samples, func(x, y int) bool {
+		sx, sy := samples[x], samples[y]
+		if sx.Timestamp != sy.Timestamp {
+			return sx.Timestamp < sy.Timestamp
+		}
+		return sx.Value < sy.Value
+	})
+}
+
+func queryblock(t testing.TB, q storage.Querier, expectedMinTime, expectedMaxTime int64, lbls labels.Labels, matchers ...*labels.Matcher) ([]backfillSample, error) {
```

Comment:
query is also declared in unittest.go in promtool so cannot rename

Reply:
Then we can change it to `queryAllSeries` and get rid of the `matchers` in the argument since we are only using it in 1 place with 1 matcher, no need to make it generic for now (since `queryblock` is not the right name here)

Reply:
Done :)